### PR TITLE
Re-align configurations in helm docs

### PR DESCRIPTION
**Pull request by shaun-nx**
### Proposed changes

Adds the below list of helm arguments to the Installation with Helm docs:

- controller.autoscaling.behavior
- controller.defaultHTTPListenerPort
- controller.defaultHTTPSListenerPort
- controller.ingressClass.setAsDefaultIngress
- controller.initContainerResources

Removes the below helm argument from the installation with Helm docs:

- controller.enablePreviewPolicies
- controller.setAsDefaultIngress

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: docs/content/installation/installing-nic/installation-with-helm.md [hunk 1]
```diff
@@ -335,15 +335,16 @@ The following tables lists the configurable parameters of the NGINX Ingress Cont
...
 | **controller.ingressClass.create** | Creates a new IngressClass object with the name `controller.ingressClass.name`. Set to `false` to use an existing ingressClass created using `kubectl` with the same name. If you use `helm upgrade`, do not change the values from the previous release as helm will delete IngressClass objects managed by helm. If you are upgrading from a release earlier than 3.3.2, do not set the value to false. | true |
+| **controller.ingressClass.setAsDefaultIngress** | New Ingresses without an `"ingressClassName"` field specified will be assigned the class specified in `controller.ingressClass.name`. Requires `controller.ingressClass.create`.  | false |
 | **controller.setAsDefaultIngress** | New Ingresses without an `"ingressClassName"` field specified will be assigned the class specified in `controller.ingressClass.name`. | false |
```
Comment (vepatel) [inline docs/content/installation/installing-nic/installation-with-helm.md:343]: needs to be removed

Comment (ADubhlaoich) [reply docs/content/installation/installing-nic/installation-with-helm.md:343]: I added this as a suggestion so Shaun could accept it without leaving the GitHub UI.

**Review by ADubhlaoich — APPROVED** [review]
LGTM on docs side, I've left an inline suggestion to match @vepatel 's feedback.

File: docs/content/installation/installing-nic/installation-with-helm.md (hunk 1 above)
Comment (shaun-nx) [reply docs/content/installation/installing-nic/installation-with-helm.md:343]: @vepatel Nice catch

File: docs/content/installation/installing-nic/installation-with-helm.md [hunk 2]
```diff
@@ -335,15 +335,16 @@ The following tables lists the configurable parameters of the NGINX Ingress Cont
...
 | **controller.ingressClass.create** | Creates a new IngressClass object with the name `controller.ingressClass.name`. Set to `false` to use an existing ingressClass created using `kubectl` with the same name. If you use `helm upgrade`, do not change the values from the previous release as helm will delete IngressClass objects managed by helm. If you are upgrading from a release earlier than 3.3.2, do not set the value to false. | true |
+| **controller.ingressClass.setAsDefaultIngress** | New Ingresses without an `"ingressClassName"` field specified will be assigned the class specified in `controller.ingressClass.name`. Requires `controller.ingressClass.create`.  | false |
```
[the remark this answers is not in the document]
Comment (vepatel) [reply docs/content/installation/installing-nic/installation-with-helm.md:342]: @ADubhlaoich this is the one which replaces the above arg I asked to be removed so no changes required here

**Review by vepatel — APPROVED** [review]

**Review by jjngx — APPROVED** [review]

**Review by oseoin — APPROVED** [review]

**Review by haywoodsh — APPROVED** [review]

**Review by j1m-ryan — APPROVED** [review]

**Review by pdabelf5 — APPROVED** [review]

**Pull request merged**